libs/wingc/src/lsp/completions.rs

```diff
@@ -91,18 +91,25 @@ pub fn on_completion(params: lsp_types::CompletionParams) -> CompletionResponse
 
 			if parent.kind() == "nested_identifier" {
 				if let Some(nearest_expr) = scope_visitor.nearest_expr {
-					let nearest_expr_type = types.get_expr_type(nearest_expr).unwrap();
-
-					// If we are inside an incomplete reference, there is possibly a type error or an anything which has no completions
-					if !nearest_expr_type.is_anything() && !nearest_expr_type.is_unresolved() {
-						return get_completions_from_type(
-							&nearest_expr_type,
-							types,
-							scope_visitor
-								.found_scope
-								.map(|s| s.env.borrow().as_ref().expect("Scopes must have an environment").phase),
-							true,
-						);
+					let parent_start = parent.start_position();
+					let parent_start_pos = lsp_types::Position {
+						line: parent_start.row as u32,
+						character: parent_start.column as u32,
+					};
+					if nearest_expr.span.contains(&parent_start_pos) {
+						let nearest_expr_type = types.get_expr_type(nearest_expr).unwrap();
+
+						// If we are inside an incomplete reference, there is possibly a type error or an anything which has no completions
+						if !nearest_expr_type.is_anything() && !nearest_expr_type.is_unresolved() {
```

Comment:
I have a feeling that in case this is an `any` we probably don't want to continue here (calling `get_completions_from_type` with its empty result is good enough). In case of `unresolved` then we probably do want to continue in this function.

Reply:
Yeah, makes sense. Changed it to only check for `unresolved`, since get_completions_from_type will return an empty list for `any` anyways.